### DISCUSSION TITLE
Convert all delete values to 0

### DISCUSSION THREAD

**Discussion by hawklorry** [DHI/mikeio#322] [Q&A]
Is there a function to covert all delete values to 0? Thanks.

**Comment by ecomodeller** [comment]
No, but we could create one like Pandas has https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.fillna.html

**Reply by jsmariegaard** [reply]
This would be really simple - it could be added to both DataArray and Dataset. And it would be consistent with both pandas and xarray.

**Comment by hawklorry** [comment]
Thanks @ecomodeller . 

The reason I'm asking is that the dfs0 file I'm using is created from a Pandas DataFrame with NULL value filledwith zero as shown below but  it still gives me the delete value warning in MIKE+. Any thoughts on that? The file is available at ftp://tea42@ftp.dhigroup.com/pub/ZHYU/intensity_5min.dfs0. 


`merged = merged.fillna(value = 0)`

**Comment by jsmariegaard** [comment]
Have you tried with boolean indexing with DataArray? As showed in https://github.com/DHI/mikeio/blob/main/notebooks/DataArray%20-%20Masking.ipynb 

Something like this? 

    da = mikeio.read("intensity_5min.dfs0")[0]
    da[da==deletevalue] = 0
    da.to_dfs0("new_file.dfs0")

**Reply by hawklorry** [reply]
Thanks @ecomodeller  and @jsmariegaard . It turns out the warning message in MIKE+ is a bug. The dfs0 file is fine.
